pyerrors/correlators.py

```diff
@@ -254,6 +254,18 @@ def is_matrix_symmetric(self):
                         return False
         return True
 
+    def trace(self):
+        """Calculates the per-timeslice trace of a correlator matrix."""
+        if self.N == 1:
+            raise TypeError("Only works for correlator matrices.")
```

Comment:
Strictly speaking, it is not necessary to catch this, since the trace of a 1x1 matrix should be well defined. But of course it does not make much sense.

Reply:
I just checked and numpy does not like the trace of an array with a single entry:
```python
np.trace(np.array([1.1]))
> ValueError: diag requires an array of at least two dimensions
```